Library/Homebrew/utils/curl.rb

```diff
@@ -118,36 +118,37 @@ def curl(*args, print_stdout: true, **options)
       result
     end
 
-    def curl_download(*args, to: nil, partial: true, **options)
+    def parse_headers(headers)
+      return {} unless headers
+
+      # Skip status code
+      headers.split("\r\n")[1..].to_h { |h| h.split(": ") }
```

Comment:
```suggestion
      headers.split("\r\n")[1..].to_h do |h|
        name, content = h.split(": ")
        [name.downcase, content]
      end
```

I think we need to downcase everything. I've seen some headers that have caps and some with lowercase (I think it depends on whether `HTTP/1.1` or `HTTP/2` is being used but I'm not an expert:

```console
$ curl -I https://example.com
HTTP/2 200
content-encoding: gzip
accept-ranges: bytes
age: 341342
cache-control: max-age=604800
content-type: text/html; charset=UTF-8
date: Fri, 14 May 2021 14:52:49 GMT
etag: "3147526947"
expires: Fri, 21 May 2021 14:52:49 GMT
last-modified: Thu, 17 Oct 2019 07:18:26 GMT
server: ECS (ord/4CD8)
x-cache: HIT
content-length: 648

$ curl -I https://go.microsoft.com/
HTTP/1.1 302 Moved Temporarily
Server: AkamaiGHost
Content-Length: 0
Location: https://www.microsoft.com/
Expires: Fri, 14 May 2021 14:54:36 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Fri, 14 May 2021 14:54:36 GMT
Connection: keep-alive
Strict-Transport-Security: max-age=31536000 ; includeSubDomains
```

Reply:
You are correct. From the HTTP/2 spec:
>Just as in HTTP/1.x, header field names are strings of ASCII characters that are compared in a case-insensitive fashion.  However, header field names MUST be converted to lowercase prior to their encoding in HTTP/2.  A request or 
response containing uppercase header field names MUST be treated as malformed (Section 8.1.2.6).

In HTTP/1.x, header names are case insensitive.

Coincidentally, your changes also fix an issue where headers with no content would cause `parse_headers` to fail. `to_h` expects an array with two items, but `split(": ")` called on a header with no content would return an array with one item.